dashboard/config/packages/old_sound_rabbit_mq.yaml

```diff
@@ -7,7 +7,7 @@ old_sound_rabbit_mq:
             callback: Libero\Dashboard\Consumer\EventsConsumer
             enable_logger: true
             exchange_options:
-                name: 'dashboard'
-                type: direct
+                name: 'articles'
+                type: fanout
```

Comment:
not sure why the dashboard should know about the type of the exchange, is it required to subscribe to it? The risk is a race condition where the dashboard preempts the article store in creating the exchange, which shouldn't belong to that service.

Reply:
Untested, but looks like you can set `declare: false` and it won't create it.

Reply:
Trying to test this now. Which level is `declare` configured at in the file?

Reply:
Same level (https://github.com/php-amqplib/RabbitMqBundle/blob/cf67adaa4e306d8e9cb6855a72d79263b425ded8/DependencyInjection/Configuration.php#L371-L389), so should be:

```yaml
exchange_options:
    name: 'articles'
    type: fanout
    declare: false
```

Reply:
Ah ok, thank you. I thought the `declare` meant we did not have to specify a value for the `type`.

OK, this works. It does not create the exchange when `declare: false` is present.

Reply:
I was surprised too, feels redundant.

Reply:
This however causes the following condition: (after a clean startup with no previous volume data)

- `dasboard` queue present
- `articles` exchange not present until the store creates it
- `dashboard` therefore cannot bind to it: (error on startup)
```
In AMQPChannel.php line 188:

NOT_FOUND - no exchange 'articles' in vhost '/'  
```
- There does not seem to be another attempt to bind (though are we expecting it to?)
- Even after a manual binding (via the rabbitMQ UI) of `dashboard` --> `articles` and new events being sent, the dashboard (app) still does not consume the events. I can however see the events pending on the queue (via the rabbitMQ UI) so they are being delivered to the `dashboard` queue.

Will probably benefit from an in person discussion.

Reply:
Yes, we've seen this before https://github.com/libero/walking-skeleton/pull/25/files#diff-bca2c15ea701ffde5b33224920caa539R39

Reply:
As discussed, I will not implement the `declare: false`, it will be handled in another pr along with a method of checking the exchange is declared before trying to bind.